src/Control/Lens/Internal/Indexed.hs

```diff
@@ -157,10 +154,20 @@ instance Profunctor (Indexed i) where
 #ifndef SAFE
   ( .# ) ibc _ = coerce ibc
   {-# INLINE ( .# ) #-}
+#if __GLASGOW_HASKELL__ < 710
```

Comment:
Hah, I didn't expect this patch to _introduce_ CPP. How unfortunate. Surely there is a way to avoid this somehow? Does the trick that the `profunctors` library use [here](https://github.com/ekmett/profunctors/blob/2ef115beef4c513b8b3ca3f1bc4abdf648182cc0/src/Data/Profunctor/Unsafe.hs#L175) also work here?

Reply:
Well, I'd leave it here and rather drop GHC-7.10 in the year or two. CPP is explicit

Reply:
Very well. But at the very least I'd like to minimize the use of CPP to one location, if possible. `coerce'` is defined four separate times in its own CPP block, and each use site of `coerce'` also uses its own CPP! Better yet to have a single definition for `coerce`', like this:

```hs
-- | Used only to work around a GHC 7.8-specific `Coercible` solver bug when implementing ('#.').
coerce' :: forall a b. Coercible a b => b -> a
#if __GLASGOW_HASKELL__ >= 710
coerce' = coerce
#else
coerce' = coerce (id :: a -> a)
#endif
{-# INLINE coerce' #-}
```

Reply:
Yes, I removed `Internal.Coercible` with a hope that it's not necessary. But seems it is.